src/EditorFeatures/CSharpTest/Diagnostics/UpgradeProject/UpgradeProjectTests.cs

```diff
@@ -0,0 +1,189 @@
+
+                var appliedChanges = ApplyOperationsAndGetSolution(workspace, operations);
+                var oldSolution = appliedChanges.Item1;
+                var newSolution = appliedChanges.Item2;
+                Assert.True(newSolution.Projects.Where(p => p.Language == LanguageNames.CSharp)
+                    .All(p => ((CSharpParseOptions)p.ParseOptions).SpecifiedLanguageVersion == expected));
+            }
+
+            await TestAsync(initialMarkup, initialMarkup, parseOptions); // no change to markup
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUpgradeProject)]
+        public async Task UpgradeProjectToDefault()
+        {
+            await TestLanguageVersionUpgradedAsync(
+@"
+class Program
+{
+    void A()
+    {
+        var x = [|(1, 2)|];
+    }
+}",
+                LanguageVersion.Default,
+                new CSharpParseOptions(LanguageVersion.CSharp6));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUpgradeProject)]
+        public async Task UpgradeProjectToCSharp7()
+        {
+            await TestLanguageVersionUpgradedAsync(
+@"
+class Program
+{
+    void A()
+    {
+        var x = [|(1, 2)|];
+    }
+}",
+                LanguageVersion.CSharp7,
+                new CSharpParseOptions(LanguageVersion.CSharp6),
+                index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUpgradeProject)]
+        public async Task UpgradeAllProjectsToDefault()
+        {
+            await TestLanguageVersionUpgradedAsync(
+@"<Workspace>
+    <Project Language=""C#"" LanguageVersion=""6"">
+        <Document>
+class C
+{
+    void A()
+    {
+        var x = [|(1, 2)|];
+    }
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" LanguageVersion=""6"">
+    </Project>
+    <Project Language=""C#"" LanguageVersion=""Default"">
+    </Project>
+    <Project Language=""C#"" LanguageVersion=""7"">
+    </Project>
+    <Project Language=""Visual Basic"">
+    </Project>
+</Workspace>",
+                LanguageVersion.Default,
+                parseOptions: null,
+                index: 2);
+
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUpgradeProject)]
+        public async Task UpgradeAllProjectsToCSharp7()
+        {
+            await TestLanguageVersionUpgradedAsync(
+@"<Workspace>
+    <Project Language=""C#"" LanguageVersion=""6"">
+        <Document>
+class C
+{
+    void A()
+    {
+        var x = [|(1, 2)|];
+    }
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" LanguageVersion=""6"">
+    </Project>
+    <Project Language=""C#"" LanguageVersion=""Default"">
+    </Project>
+    <Project Language=""C#"" LanguageVersion=""7"">
+    </Project>
+    <Project Language=""Visual Basic"">
+    </Project>
+</Workspace>",
+                LanguageVersion.Default,
+                parseOptions: null,
+                index: 2);
```

Comment:
I must admit some constants or an enum would make this less painful. Woe to the person who wants to reorder those.